javascript/react/src/components/Codebook/CodebookFieldEditor.types.ts

```diff
@@ -0,0 +1,31 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import type { Field } from '@datashaper/schema'
+
+export enum CodebookFields {
+	DisplayName = 'display name',
+	Description = 'description',
+	DataType = 'data type',
+	DataNature = 'data nature',
+	Units = 'units',
+	Mapping = 'mapping',
+}
+
+export interface CodebookFieldEditorProps {
```

Comment:
Add a styles block here so the user can customize whatever they need. There is a lot of hard-coded styling such as the widths, which may need overridden. essex-toolkit has a bunch of examples that I enabled deeper customization in so we could use them in DataShaper.

This would also provide a clean mechanism for editing the cell borders in single-versus-table mode. The default could be to have no cell border to reduce clutter, but have a cell style option, and the table editor could pass in the style to add the borders so it then looks like a table layout.

https://github.com/microsoft/essex-toolkit/blob/main/packages/components/src/ToolPanel/ToolPanel.types.ts